### PR TITLE
Sync blackbox log viewer with BF 3.3

### PR DESCRIPTION
**Pull request by basdelfos**
Sync blackbox log viewer with BF 3.3

- Fixed header dialog
- Fix loading logs with odd blackbox rates (e.g. 1.5k) --> this seems to be fixed, probably already been fixed with #106 
- Added separate field for rates and deadband.
image

### CODE REVIEW

File: js/header_dialog.js [hunk 1]
```diff
@@ -58,7 +58,9 @@ function HeaderDialog(dialog, onSave) {
 	        {name:'setpointRelaxRatio'			, type:FIRMWARE_TYPE_BETAFLIGHT,  min:'3.1.0', max:'999.9.9'},
             {name:'antiGravityGain'             , type:FIRMWARE_TYPE_BETAFLIGHT,  min:'3.1.0', max:'999.9.9'},
 	        {name:'antiGravityThreshold'        , type:FIRMWARE_TYPE_BETAFLIGHT,  min:'3.1.0', max:'999.9.9'},
-            {name:'itermWindupPointPercent'     , type:FIRMWARE_TYPE_BETAFLIGHT,  min:'3.1.0', max:'999.9.9'}
+			{name:'itermWindupPointPercent'     , type:FIRMWARE_TYPE_BETAFLIGHT,  min:'3.1.0', max:'999.9.9'},
```
Comment (mikeller) [inline js/header_dialog.js]: Tabs instead of spaces.

File: js/header_dialog.js [hunk 2]
```diff
@@ -208,30 +210,35 @@ function HeaderDialog(dialog, onSave) {
...
             {bit: 13, group: 'rxMode', mode: 'group', name: 'RX_PARALLEL_PWM', description: 'PWM receiver selected'},
             {bit: 14, group: 'rxMode', mode: 'group', name: 'RX_MSP', description: 'Controller over MSP'},
             {bit: 15, group: 'other', name: 'RSSI_ADC', description: 'ADC RSSI Monitoring'},
             {bit: 16, group: 'other', name: 'LED_STRIP', description: 'Addressible RGB LED strip support'},
-            {bit: 17, group: 'other', name: 'DISPLAY', description: 'OLED Screen Display'},
-            {bit: 19, group: 'other', name: 'BLACKBOX', description: 'Blackbox flight data recorder'},
+			{bit: 17, group: 'other', name: 'DISPLAY', description: 'OLED Screen Display'},
             {bit: 20, group: 'other', name: 'CHANNEL_FORWARDING', description: 'Forward aux channels to servo outputs'},
-            {bit: 21, group: 'other', name: 'TRANSPONDER', description: 'Transponder enabled'}
+			{bit: 21, group: 'other', name: 'TRANSPONDER', description: 'Race Transponder'},
```
Comment (mikeller) [inline js/header_dialog.js]: Tabs.

File: js/flightlog_parser.js [hunk 3]
```diff
@@ -483,7 +490,14 @@ var FlightLogParser = function(logData) {
                 that.sysConfig.rcYawRate = parseInt(fieldValue, 10);
                 break
             case "rc_expo":
-                that.sysConfig.rcExpo = parseInt(fieldValue, 10);
+                if(fieldValue.match(/.*,.*/)!=null) {
```
Comment (mikeller) [inline js/flightlog_parser.js]: What does `parseCommaSeparatedString()` return if it's not a comma separated string?

It's more robust to use the `parseCommaSeparatedString()` function to check if it matches (and return null if not), than to do the checking here and the parsing in the function.

**Comment by etracer65** [comment]
In `js/flightlog.js` the function `rcCommandRawToDegreesPerSecond()` needs to be changed to deal with the revised `rc_rates` and `rc_expo` array contents.  There will have to be a (Betaflight >= 3.3) or (and Cleanflight > 2.3) section added I'm guessing.

File: js/flightlog_parser.js (hunk 3 above)
Comment (basdelfos) [reply js/flightlog_parser.js]: I've reused that check from somewhere else in the code.

```
function parseCommaSeparatedString(string, length) {
    /***
     * Parse a comma separated string for individual values.
     *
     * string               is the comma separated string to parse
     * length (optional)    the returned array will be forced to be this long; extra fields will be discarded,
     *                      missing fields will be padded. if length is not specified, then array will be auto
     *                      sized.
     *
     * returns              if the string does not contain a comma, then the first integer/float/string is returned
     *                      else an Array is returned containing all the values up to the length (if specified)
     ***/
```

The result of `parseCommaSeparatedString()` is never `null`, it always returns something. I guess that is why the check done separately.

I can make this nicer to do the check in function instead.

**Comment by etracer65** [comment]
The changes to the rate calculation look good to me.  The only other thing I see is a minor issue in the log header display where roll/pitch are not separated in >= 3.3.

blackbox_header

**Comment by basdelfos** [comment]
This PR is ready to be merged.

File: js/flightlog.js [hunk 4]
```diff
@@ -980,12 +979,30 @@ FlightLog.prototype.rcCommandRawToDegreesPerSecond = function(value, axis, curre
...
+                    case AXIS.PITCH:
+                        rcExpo = sysConfig["rc_expo"][1];
+                        rcRate = sysConfig["rc_rates"][1] / 100.0;
+                        break;
+                    case AXIS.YAW:
+                        rcExpo = sysConfig["rc_expo"][2];
+                        rcRate = sysConfig["rc_rates"][2] / 100.0;
+                        break;
+                }
+            }
```
Comment (mikeller) [inline js/flightlog.js]: Stray newline.

**Comment by mikeller** [comment]
@basdelfos: Thanks for replacing the tabs with spaces. For next time, it is better to do these global replaces in a pull request of their own, since reviewing files that contain 100s of whitespace changes along with a few code changes is very cumbersome.

**Pull request merged**